R/detect_column.r

```diff
@@ -0,0 +1,354 @@
+  dat1$Variable <- as.character(dat1$Variable)
+  dat1$Type <- as.character(dat1$Type)
+  names(domain_spec)[names(domain_spec)=='Variable Name'] <- 'Variable'
+  domain_spec_type <- select(domain_spec,Variable,Type)
+  # dat1$Type[dat1$Type %in% c('integer','numeric','logical')] <- 'Num'
+  dat1$Type[dat1$Type == 'integer'] <- 'Num'
+  dat1$Type[dat1$Type == 'numeric'] <- 'Num'
+  dat1$Type[dat1$Type == 'logical'] <- 'Num'
+  dat1$Type[dat1$Type == 'character'] <- 'Char'
+  dat1 <- left_join(dat1,domain_spec_type,by='Variable')
+  names_num_char <- dat1$Variable[dat1$Type.x == 'Num' & dat1$Type.y== 'Char']
+  names_num_char <- names_num_char[is.na(names_num_char)==F]
+  
+  if(length(names_num_char)>0){
+  warning(
+    "The following columns of class 'numeric' or 'logical' should be and have been converted to character: ",
+    paste(names_num_char[is.na(names_num_char)==F], collapse=", ")
+  )
+  }
+  
+  names_char_num <- dat1$Variable[dat1$Type.x == 'Char' & dat1$Type.y== 'Num']
+  names_char_num <- names_char_num[is.na(names_char_num)==F]
+  
+  if(length(names_char_num)>0) {
+  warning(
+    "The following columns of class 'character' should be and have been converted to numeric: ",
+    paste(names_char_num[is.na(names_char_num)==F], collapse=", ")
+  )
+  }
+  
+  data_sub <- data
+  
+  if(length(names_num_char)>0){
+  for (i in names_num_char){
+    data_sub[[i]] <- as.character(data[[i]])
+  }
+  }
+  
+  if(length(names_char_num)>0) {
+  for (j in names_char_num){
+    data_sub[[j]] <- as.numeric(data[[j]])
+  }
+  }
+  # Modify input to data at global level
+  # data_name <- deparse(substitute(data))
+  # assign(data_name,data_sub,envir=parent.frame())
+  
+  na_names <- list()
+  for(i in names(data)){
+    if(any(is.na(data[[i]]))){
+      na_names[[i]] <- i
+    }
+  }
+  na_names <- unlist(na_names)
+  if(length(na_names) > 0) {
+  stop(
+    "The following columns in dataset have NA: ",
+    paste(na_names, collapse=", ")
+  )
+  }
+  data <- data_sub
+  data
+}
+
+# validate_column_class(dat,domain_spec=domain_specs$DM)
+
+#' Check to see if columns have controlled terminology, give an error if columns do not have controlled terminology.
+#' 
+#' @param data The dataset whose columns to check.
+#' @param domain_spec The domain specification ("CO", "DM", "SE", "SM", "SV") for checking columns.
+#' @return If columns contain do not have controlled terminology, give an error and indicate which are those columns.
+#' @export
+validate_ctrl_terminology <- function(data,domain_spec){
+  if(domain_spec=="CO"){
+    data("CO")
+    domain_spec <- force(CO)
+  } else if (domain_spec=="DM"){
+    data("DM")
+    domain_spec <- force(DM)
+  } else if (domain_spec=="SE"){
+    data("SE")
+    domain_spec <- force(SE)
+  } else if (domain_spec=="SM"){
+    data("SM")
+    domain_spec <- force(SM)
+  } else if (domain_spec=="SV"){
+    data("SV")
+    domain_spec <- force(SV)
+  }
```

Comment:
Could this be made simpler by accessing a named list of domain_spec values like:

`domain_spec_current <- domain_spec_list[[sdtmig_version]][[domain_name]]`